src/python/pants/engine/target.py

```diff
@@ -2364,6 +2364,28 @@ def to_path_globs(
             for globs in self.value
         )
 
+    def flatten(self) -> dict[str, dict[str, Any]]:
```

Comment:
It'd be great to add the tests from here: https://github.com/pantsbuild/pants/pull/13187/files#diff-69ab3877fc85a5b1fdaf9bb56cd41a981f63754e10375d062107bd4df06eb3c5R1178

Reply:
ahh yes!